neuro_san/service/main_loop/server_main_loop.py

```diff
@@ -181,30 +205,33 @@ def main_loop(self):
                     self.manifest_update_period_seconds,
                     self.server_status)
             updater.start()
-        else:
-            # We don't use manifest updater,
-            # so set it to "ready" not to mess up overall status.
-            self.server_status.set_updater_status(True)
-
-        # Start HTTP server side-car:
-        http_server = HttpServer(
-            self.server_status,
-            self.http_port,
-            self.service_openapi_spec_file,
-            self.request_limit,
-            self.network_storage_dict,
-            forwarded_request_metadata=metadata_str)
+
+        if self.server_status.http_service.is_requested():
+            # Create HTTP server;
+            self.http_server = HttpServer(
+                self.server_status,
+                self.http_port,
+                self.service_openapi_spec_file,
+                self.request_limit,
+                self.network_storage_dict,
+                forwarded_request_metadata=metadata_str)
 
         # Now - our servers (gRPC and http) are created and listen to updates of network_storage
         # Perform the initial setup
         public_storage: AgentNetworkStorage = self.network_storage_dict.get("public")
         public_storage.setup_agent_networks(self.agent_networks)
 
         # Start all services:
-        http_server_thread = threading.Thread(target=http_server, args=(self.grpc_server,), daemon=True)
-        http_server_thread.start()
+        http_server_thread = None
+        if self.server_status.http_service.is_requested():
+            http_server_thread = threading.Thread(target=self.http_server, args=(self.grpc_server,), daemon=True)
+            http_server_thread.start()
+
+        if self.server_status.grpc_service.is_requested():
+            self.grpc_server.serve()
 
-        self.grpc_server.serve()
+        if http_server_thread is not None:
+            http_server_thread.join()
 
```

Comment:
Wait for http service thread, otherwise we can prematurely finish the whole server process.